mesonbuild/backend/backends.py

```diff
@@ -517,7 +517,11 @@ def generate_basic_compiler_args(self, target, compiler, no_warn_args=False):
             # For 'automagic' deps: Boost and GTest. Also dependency('threads').
             # pkg-config puts the thread flags itself via `Cflags:`
             if dep.need_threads():
-                commands += compiler.thread_flags(self.environment)
+                try:
+                    commands += compiler.thread_flags(self.environment)
+                except AttributeError:
+                    mlog.debug('{!r} {!r} compiler does not support thread_flags. '
+                               'Ignored.'.format(compiler.get_id(), compiler.get_language()))
```

Comment:
I think a better way to solve this is to implement `compiler.thread_flags()` in the `Compiler` base class to always return `[]`. That way classes that inherit it can override it.

Reply:
Should I also add `thread_link_flags()` stub?

Reply:
Hmm, no, that would be used by the linker, and is likely to mask bugs in our linker selection or cause linker errors due to undefined thread symbols.